Insert button

iantance opened the pull request:
@djreimer Ready for review

Addresses DripEmail/drip#4001

This adds the button plugin and allows us to pass "class" and "removeClass" properties with the "updateLink" event.

derrickreimer commented on plugins/driplink/plugin.js:
```diff
@@ -78,6 +78,26 @@
+					} else
+						element = null;
+
+					editor.fire("linkDialogRequested", { el: element, button: true });
```
Let's change this to `buttonDialogRequested` and remove the `button: true`. Then in our application JS, we can key off of this different event name instead of snooping around in the properties to see if it's a button request or ordinary link.

derrickreimer commented on plugins/driplink/plugin.js:
```diff
@@ -91,6 +111,11 @@
+	        label: 'Insert Button',
+	        command: 'dripbutton',
+	        toolbar: 'help'
+	      });
```
Looks like indentation is a little wonky here.

derrickreimer commented on plugins/driplink/plugin.js:
```diff
@@ -91,6 +111,11 @@
 					command: 'unlink',
 					toolbar: 'links,20'
 				} );
+	      editor.ui.addButton( 'DripButton', {
```
Not sure we need to namespace here with `Drip`, since the others are simply called "Link" and "Unlink". Let's call it `ButtonLink` instead, the a command of `buttonlink`. Also, looks like `toolbar` is `help` here while the others are `links,...`. Any particular reason for that?

iantance commented:
@djreimer I revised and renamed things on this one. I changed that `toolbar` setting to match the "Link" and "Unlink" button. I had it set differently because at one point I thought that would give us more control over the placement and appearance of the button. I fiddled around some more, and I now think we can handle that all on in the Drip application JS.
